app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala

```diff
@@ -264,12 +283,14 @@ class IncomeUpdateCalculatorController @Inject()(
         (formData: HoursWorkedForm) => {
           for {
             id <- journeyCacheService.mandatoryValueAsInt(UpdateIncome_IdKey)
-          } yield
+          } yield {
+            journeyCacheService.cache(UpdateIncome_WorkingHoursKey, formData.workingHours.getOrElse(""))
```

Comment:
Move this into the for comprehension with `_ <-` for the assignment. That way it will ensure the value is always cached as above